client-spark/spark-3/src/main/scala/org/apache/spark/shuffle/celeborn/CelebornShuffleReader.scala

```diff
@@ -60,6 +60,10 @@ class CelebornShuffleReader[K, C](
 
     val serializerInstance = newSerializerInstance(dep)
 
+    val shuffleId = shuffleClient.getShuffleId(handle.shuffleId)
+    logDebug(
+      s"get shuffleId $shuffleId for appShuffleId ${handle.shuffleId} attemptNum ${context.stageAttemptNumber()}")
```

Comment:
In theory, there is a potential race condition here (task was earlier attempt is reading from newer shuffle id) - but should be beneign imo given it is for reads - writes is where it will be an issue (if code evolves to use `getShuffleId` for that).

Reply:
I agree
For DETERMINATE stage, read shuffle data from previous stage attempt will not introduce any issue
For INDETERMINATE stage, spark aborts the job now